### ISSUE TITLE
Suggestion: add call of trim() to this line

### ISSUE THREAD

**Issue by derroman** [jamessimone/apex-dml-mocking#1]
Hi James,
awesome work on this repo,

I've recently played around specificly with this Query.cls class
https://github.com/jamessimone/apex-dml-mocking/blob/4ba09503b440aa2496dda0e43fce75db9acc7e50/force-app/repository/Query.cls#L97

And I used it to create a where clause with a Set of Ids to finally get a query like this:

```sql
SELECT 
	Id,
	Name
FROM Account  
WHERE Id IN ('0010O000028pTbxQAE','0011l00000jpAV3AAM','0013X00002fmFN1QAM')
```

But I discovered, that the method mentioned puts a leading space in front of the second and all following parts of the list (for whatever reason), so the query looks like this:

```sql
SELECT 
	Id,
	Name
FROM Account  
WHERE Id IN ('0010O000028pTbxQAE',' 0011l00000jpAV3AAM',' 0013X00002fmFN1QAM')
```
This obviously leads to an error: "System.QueryException: invalid ID field"

So what I've done in my modification of your code - I've added a call to trim().
```apex
 return '\'' + String.escapeSingleQuotes(input).trim() + '\''; 
```
What's your opinion? Maybe trim() is just a quick fix?

Source: [Apex Developer Guide String Class trim](https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_methods_system_string.htm#apex_System_String_trim)

**Comment by jamessimone** [comment]
@derroman ah, interesting find! My apologies for not catching this, the repo has a slightly different version of Query than I use in prod but that’s my fault for not checking the quote wrapping for Ids. Your fix is a good one - I will incorporate it shortly!

**Comment by derroman** [comment]
But just for fun and because I did not find the time yet (a.k.a "too lazy to debug" ;-) ):

Why does this happen in the first place?

**Comment by jamessimone** [comment]
@derroman this is now resolved. This was an interesting opportunity to delve into `toString()`'s base implementation - apparently, calling `toString()` on a Set adds spaces between the different entries; it's possible that the List class does this as well, but there it is obfuscated by an oddity singular to Set - it is an Iterable (you can use it in a `for` loop, after all), but it does not publicly show that. Most objects you can call `instanceof` to detect what they are, which is how the string parsing in the `Query` class works. However, from my previous work on this (as part of the [Nebula Framework](https://github.com/jongpie/NebulaFramework)), I knew that Sets do _not_ test true for being an Iterable, which is why they are handled separately to begin with.

Long story short - I shot myself in the foot by having a test involving Sets with multiple Ids ... but the Ids that I was using had the same value. I was able to easily reproduce your issue in the existing test that I have [purely by updating the test to include a second unique Id](https://github.com/jamessimone/apex-dml-mocking/blob/master/force-app/repository/RepositoryTests.cls#L15). That was a tragic oversight on my part; it was fine to use the same Id for testing that Lists worked with this newer implementation, but of course the same value being added to a Set just reduces it to a single value.

Thanks for bringing this to my attention, and for using this framework!

**Comment by derroman** [comment]
@jamessimone great and thanks for the interesting explanation!
